src/net/data_stream.rs

```diff
@@ -15,14 +36,25 @@ use miniconf::MiniconfAtomic;
 use serde::Deserialize;
 use smoltcp_nal::embedded_nal::{IpAddr, Ipv4Addr, SocketAddr, UdpClientStack};
 
-use super::NetworkReference;
 use crate::hardware::design_parameters::SAMPLE_BUFFER_SIZE;
 
-// The number of data blocks that we will buffer in the queue.
-const BLOCK_BUFFER_SIZE: usize = 30;
+use heapless::pool::{Box, Init, Pool, Uninit};
+
+use super::NetworkReference;
+
+const MAGIC_WORD: u16 = 0x057B;
+
+const HEADER_SIZE: usize = 8;
```

Comment:
I want to avoid usage of `repr(packed)` due to all of the unsoundness discussed around it. We need to think carefully about alignment of the structure or we could cause unexpected hardfault's of the CPU due to unaligned access. As such, I think it's easier and less painful to just define this as a static value.